inference-engine/tests/functional/plugin/shared/src/single_layer_tests/detection_output.cpp

```diff
@@ -0,0 +1,151 @@
+             inShapes[idxLocation], inShapes[idxConfidence], inShapes[idxPriors], inShapes[idxArmConfidence], inShapes[idxArmLocation]) = specificAttrs;
+
+    for (size_t i = 0; i < inShapes.size(); i++) {
+        if (inShapes[i].empty())
+            break;
+        inShapes[i][0] = batch;
+    }
+
+    attrs.nms_threshold = nmsThreshold;
+    attrs.confidence_threshold = confidenceThreshold;
+    attrs.objectness_score = objectnessScore;
+
+    std::ostringstream result;
+    result << "IS = { ";
+    result << "LOC=" << CommonTestUtils::vec2str(inShapes[0]) << "_";
+    result << "CONF=" << CommonTestUtils::vec2str(inShapes[1]) << "_";
+    result << "PRIOR" << CommonTestUtils::vec2str(inShapes[2]) << "_";
+    result << "ARM_CONF" << CommonTestUtils::vec2str(inShapes[3]) << "_";
+    result << "ARM_LOC" << CommonTestUtils::vec2str(inShapes[4]) << " }_";
+
+    result << "Classes=" << attrs.num_classes << "_";
+    result << "backgrId=" << attrs.background_label_id << "_";
+    result << "topK="  << attrs.top_k << "_";
+    result << "varEnc=" << attrs.variance_encoded_in_target << "_";
+    result << "keepTopK=" << CommonTestUtils::vec2str(attrs.keep_top_k) << "_";
+    result << "codeType=" << attrs.code_type << "_";
+    result << "shareLoc=" << attrs.share_location << "_";
+    result << "nmsThr=" << attrs.nms_threshold << "_";
+    result << "confThr=" << attrs.confidence_threshold << "_";
+    result << "clipAfterNms=" << attrs.clip_after_nms << "_";
+    result << "clipBeforeNms=" << attrs.clip_before_nms << "_";
+    result << "decrId=" << attrs.decrease_label_id << "_";
+    result << "norm=" << attrs.normalized << "_";
+    result << "inH=" << attrs.input_height << "_";
+    result << "inW=" << attrs.input_width << "_";
+    result << "OS=" << attrs.objectness_score << "_";
+    result << "TargetDevice=" << targetDevice;
+    return result.str();
+}
+
+void DetectionOutputLayerTest::Infer() {
+    inferRequest = executableNetwork.CreateInferRequest();
+    inputs.clear();
+
+    size_t it = 0;
+    for (const auto &input : cnnNetwork.getInputsInfo()) {
+        const auto &info = input.second;
+
+        InferenceEngine::Blob::Ptr blob;
+        if (it == 2) {
+            blob = make_blob_with_precision(info->getTensorDesc());
+            blob->allocate();
+            if (attrs.normalized) {
+                CommonTestUtils::fill_data_random_float<InferenceEngine::Precision::FP32>(blob, 1, 0, 100);
+            } else {
+                CommonTestUtils::fill_data_random<InferenceEngine::Precision::FP32>(blob, 10, 0, 1);
+            }
+        } else if (it == 1 || it == 3) {
+            blob = make_blob_with_precision(info->getTensorDesc());
+            blob->allocate();
+            CommonTestUtils::fill_data_random_float<InferenceEngine::Precision::FP32>(blob, 1, 0, 1000);
+        } else {
+            blob = make_blob_with_precision(info->getTensorDesc());
```

Comment:
`91-93` and `95-98` strings are duplicated instead of constant. Looks like constant can be redefined in the condition. and blob allocation will be implemented after it.

Reply:
fixed